### PR TITLE
expose the mount manager in the public api

### PR DESCRIPTION
**Pull request by icewind1991**
Part of the work of bringing #12086 to master

cc @DeepDiver1975 @PVince81

### CODE REVIEW

File: lib/public/files/mount/imountmanager.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCP\Files\Mount;
+
+interface IMountManager {
```
Comment (PVince81) [inline lib/public/files/mount/imountmanager.php]: PHPDoc that explains what it is ?

File: lib/public/files/mount/imountmanager.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	public function addMount(IMountPoint $mount);
+
+	/**
+	 * @param string $mountPoint
+	 */
+	public function removeMount($mountPoint);
+
+	/**
+	 * @param string $mountPoint
+	 * @param string $target
```
Comment (PVince81) [inline lib/public/files/mount/imountmanager.php]: PHPDoc

File: lib/public/files/mount/imountmanager.php (hunk 1 above)
Comment (PVince81) [inline lib/public/files/mount/imountmanager.php]: Maybe also add `@since 8.2.0`

**Comment by icewind1991** [comment]
Added more phpdoc

**Comment by PVince81** [comment]
@icewind1991 minds also fixing this ?
I had this warning recently, forgot where. Just wrote it down for 8.2:
```
Argument 1 passed to OC\Files\Mount\Manager::addMount() must be an instance of OC\Files\Mount\MountPoint, instance of Mock_IMountPoint_f1c5ecfe given
```

**Comment by karlitschek** [comment]
makes sense :+1:

**Comment by icewind1991** [comment]
> Argument 1 passed to OC\Files\Mount\Manager::addMount() must be an instance of OC\Files\Mount\MountPoint, instance of Mock_IMountPoint_f1c5ecfe given

I believe https://github.com/owncloud/core/commit/57945dbc4b0144bdebeda5cf15a168556118e9e3 will fix that

**Comment by PVince81** [comment]
Thanks. I'll regression-test this later.
The code change looks good.

**Comment by PVince81** [comment]
:+1: sharing, external storage still work

**Pull request merged**